### ISSUE TITLE
output_shape doesn't make sense, the output remains no structure

### ISSUE THREAD

**Issue by CaptainDuke** [tensorflow/io#342]
As mentioned before, I tried to read hdf5 data using tf-io, but the output_shape doesn't make sense.  ([1,8]) expected but returns ( ).

For comparing, I worte dataset2 form tf.data.Dataset.from_tensor_slices(). Also I did some tf.reshape ops, where dataset read via tf-io failed. 

I thinks it's a kind of serious problem since we can not continue tf.layers.conv2d or other ops without output_shapes.

Hope this issue would be fixed. Thanks.

![image](https://user-images.githubusercontent.com/24515303/61100918-897b7880-a49a-11e9-8bc2-a3c75db282d7.png)
![image](https://user-images.githubusercontent.com/24515303/61100992-c47dac00-a49a-11e9-905a-94619b9216c5.png)

**Comment by CaptainDuke** [comment]
Code above 
[issue_test.py.zip](https://github.com/tensorflow/io/files/3384898/issue_test.py.zip)

**Comment by CaptainDuke** [comment]
temporary solution:
add dataset.batch() and then tf.reshape
`
dataset = hdf5_io.HDF5Dataset(
       [filename], columns, output_types, output_shapes)
dataset = dataset.batch(10)
iterator = dataset.make_one_shot_iterator()
get_next = iterator.get_next()

 input_x = tf.reshape(get_next, output_shape)
sess = tf.Session()
vv = sess.run(tf.shape(input_x))
print(vv)
`

Besides, if we want to use dataset.shuffle and dataset.prefetch then, dataset.make_one_shot_iterator() works.

**Comment by yongtang** [comment]
@CaptainDuke in 2.0 the output shape and dtype are implicitly probed. I believe the issue has been resolved. Will close the issue for now but feel free to reopen if issue still persists.
